### PR TITLE
Refactor api route and subscription form to use server actions/app router

### LINKED ISSUE #1362: [Task]: Move API Handlers to newer app router "Route Handlers"
### Summary

Both the health check and Sendy API route are using the page router "API Handler".

We should move both to Route Handlers : https://nextjs.org/docs/app/building-your-application/routing/route-handlers

### Acceptance criteria

- Sendy API route working 
- Health check API working
- Tests passing

### PR DESCRIPTION
**Pull request by emilycnava**
## Summary
Note: This is still a draft, feel free to look over if you're curious, but this is not ready for merging, the main form functionality has all been completed and is on feature parity with the old form, however less than 33% of the tests are currently passing. Testing the server actions adequately via mocks with jest and playwright is proving difficult and documentation minimal/non-existent with some online sources even suggesting 'don't bother integration testing it' or approx. that advice. Dev on this branch is on pause until I can touch base with Aaron once he's feeling better and back to see if he has any insight into some of these testing errors.

Fixes #1362 

TLDR: Refactors the last remaining remnant of pages route to use the new app router paradigm of server actions for form submissions.

## Changes proposed
- Add zod for server side verification, more robust than regex match used previously
- Refactor form to use server actions (https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations)
- Translate the SENDY code from the old pages api route to a server action
- Ensure exhaustive i18n use (previous form had inconsistent i18n usage)
- Improve testing (aiming for full test coverage including form validation error states)

## Additional information
Screenshot 2024-10-09 at 12 11 20 PM

### CODE REVIEW

**Comment by acouch** [comment]
I was not able to get the unit test working. I was able to suppress the `Not implemented: HTMLFormElement.prototype.requestSubmit` but the issue seems that [jest doesn't work with form actions AFAICT](https://github.com/vercel/next.js/issues/54757).

<details>
<summary>suppress requestSubmit warning</summary>
```javascript
  beforeEach(() => {
    const origErrorConsole = window.console.error;
  
    window.console.error = (...args: any []) => {
      const firstArg = args.length > 0 && args[0].toString();
      console.log(firstArg, typeof firstArg);
  
      const shouldBeIgnored =
        firstArg &&
        typeof firstArg === 'string' &&
        firstArg.includes('Not implemented: HTMLFormElement.prototype.requestSubmit');
  
      if (!shouldBeIgnored) {
        origErrorConsole(...args);
      }
    };
  });
</details>


I'm going to move on to try and unit test the server action and create an e2e test.

**Comment by emilycnava** [comment]
Closing in favor or https://github.com/HHS/simpler-grants-gov/pull/2530
